plugins/module_utils/azure_rm_common.py

```diff
@@ -395,7 +395,8 @@ class AzureRMModuleBase(object):
     def __init__(self, derived_arg_spec, bypass_checks=False, no_log=False,
                  check_invalid_arguments=None, mutually_exclusive=None, required_together=None,
                  required_one_of=None, add_file_common_args=False, supports_check_mode=False,
-                 required_if=None, supports_tags=True, facts_module=False, skip_exec=False, is_ad_resource=False):
+                 required_if=None, supports_tags=True, facts_module=False, skip_exec=False,
+                 is_ad_resource=False, required_by={}):
```

Comment:
Dangerous default value {} as argument
```suggestion
                 is_ad_resource=False, required_by=None):
                 
      if not required_by:
          required_by = dict()
```